### PR TITLE
Fix #25805: Containers Portlet only show containers from current site

### LINKED ISSUE #25805: [Containers] : Only show containers from current site
### Parent
#23140

### Problem Statement
In the containers porlet we are listing containers from all site.

### Steps to Reproduce
1. Log in to dotCMS
2. Create 2 sites
3. Add a container to second site
4. Navigate to the New Containers Portlet (http://localhost:8080/dotAdmin/#/containers)
5. Notice that containers from all sites are listed

### Acceptance Criteria
The Containers Portlet should only list containers from the current site.

### External Links
N/A

### Assumptions & Initiation Needs
The containers endpoint should take the site id.

### Quality Assurance Notes & Workarounds
N/A

### LINKED ISSUE #25809: [Containers] : Fix Issue with Blank Containers After Filtering
### Parent
#23140

### Problem Statement
In the containers when I select a content type to filter and there is no containers with that filter it shows blank instead of the "No results found" message.

### Steps to Reproduce
https://github.com/dotCMS/core/assets/751424/c7751c7b-63bf-48c6-b2ed-b5605fe47499

### Acceptance Criteria
When a user selects a content type to filter containers in dotCMS and there is no containers that match the filter, the page should show a "No results found" message instead of blank

### External Links
N/A

### Assumptions & Initiation Needs
- It is assumed that this bug is reproducible in the latest version of dotCMS
- No additional resources are needed to initiate this bug

### Quality Assurance Notes & Workarounds
- N/A

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
🎨🚀✅

Refactored the container list component to use the `p-table` component and the `DotContainerListStore` service for better UI and performance. Fixed a z-index issue with the content type selector component. Added a method and a test case for resetting extra parameters in the paginator service.

> _We're breaking free from the old data table_
> _We're using the `p-table` to unleash our power_
> _We're refactoring the code with the `DotContainerListStore`_
> _We're fixing the bugs and improving the performance_

### Walkthrough
*  Replace `dot-listing-data-table` component with `p-table` component for better performance and flexibility ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-04da8afcab5e4883bc097456e55afa6dcedc2bf85aa55facc134e3ccb4656968L69-R112), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-04da8afcab5e4883bc097456e55afa6dcedc2bf85aa55facc134e3ccb4656968L89-R137), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L141-L163), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L308-R331), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L405-R505), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL95-R132), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL130-R182), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL166-R199), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL13-R15), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL30-R32), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL39-R43))
*  Move `dot-content-type-selector` component and `div` with the `container-listing__header-options` class to the `dot-action-header` component for common header actions ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-04da8afcab5e4883bc097456e55afa6dcedc2bf85aa55facc134e3ccb4656968L89-R137), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL13-R15), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL39-R43))
*  Add `container-list` component styles for the layout and responsiveness of the `p-table` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-622b2e167b33633f74a3961bb864f2dad8cb0ec49375b838068a4c19038bc74dR3-R27), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-622b2e167b33633f74a3961bb864f2dad8cb0ec49375b838068a4c19038bc74dR48-R54))
*  Add `DotContainerListStore` service to manage the state and effects of the containers list ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L4-R32), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L39-R55), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L51-R64), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528R227-R229), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L244-R264), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528R303-R307), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L372-R383), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL23), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL32-R47), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL45-R66), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL63-R80), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL95-R132), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL130-R182), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL166-R199), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R2), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349L8-R9), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349L14-R24), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R42-R44), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R53-R54), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349L52-R115), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349L88-R128), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349L96-R147), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R171-R177), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R202-R282), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-033a32c658da882cbc2420b75ca50850898c3f1e6c722bb2add6522da75c6349R685-R721))
*  Add `ElementRef`, `QueryList`, `ViewChild`, `ViewChildren` and `Menu` imports for the Angular decorators and types for the `container-list` component and the `p-table` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL3-R21))
*  Add `InputTextModule` and `TableModule` imports for the modules needed for the `p-table` component and the query input ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL7-R9), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL30-R32), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL39-R43))
*  Add `ActionHeaderModule` import for the module needed for the `dot-action-header` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL13-R15), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-52692359e0a7cfc5ec5fc38c1cc614a1791807693c32ccd3dc0bd25c774bf12aL39-R43))
*  Add `actionsMenu`, `tableRows`, `selectedContainers` and `destroy$` properties for the references and variables for the `container-list` component and the `p-table` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL32-R47))
*  Add `handleArchivedFilter`, `handleQueryFilter`, `loadDataPaginationEvent`, `handleActionMenuOpen` and `focusFirstRow` methods for the query filter, the pagination, the action menu and the keyboard navigation for the `p-table` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL95-R132))
*  Add `appendTo="body"` attribute to the `p-dropdown` element to fix a z-index issue with the dropdown menu ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-a5cd33958e34508a8793c208ff6c6b4d6a6692fe9b44b39d7ee6e102e1ae8cf3R8))
*  Add `resetExtraParams` method to the `PaginatorService` class to remove all extra parameters of the eventual request ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-a3a30d6f3fb50fe0e9f2a73be24e6004ba2ec00bd5f9e5cdff7ac7883114b59fR56-R64), [link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-7d2c60e5812dcabab999c813cec39b90bdb61a6c713820fb34b620307f4e6f34R116-R124))
*  Remove `DotRouterService` import from the `container-list` component since it is no longer needed with the `DotContainerListStore` service ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL23))
*  Remove `columnsMock` variable from the `container-list` component spec since it is no longer needed with the `p-table` component ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-5ea730e047e6a20e38d545ed09af293444ba1e3735f56cc02ad6c677a66c6528L141-L163))
*  Remove `getContainersWithDisabledEntities` method from the `container-list` component since it is no longer needed with the `DotContainerListStore` service ([link](https://github.com/dotCMS/core/pull/25966/files?diff=unified&w=0#diff-45fb91bed78767eafd15abea65852903141e00b936a15d2594bf8249445dca7fL130-R182))

### Screenshots

https://github.com/dotCMS/core/assets/63567962/0a7bcb96-4d19-422c-a92a-2bfecc020c7e

### CODE REVIEW

**Review by manuelrojas — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html [hunk 1]
```diff
@@ -1,63 +1,96 @@
...
+        [rows]="40"
+        [paginator]="true"
+        [pageLinks]="vm.maxPageLinks"
+        [totalRecords]="vm.totalRecords"
+        [responsiveLayout]="'scroll'"
+        (onPage)="loadDataPaginationEvent($event)"
+        selectionMode="multiple"
+        loadingIcon="pi pi-spin pi-spinner"
+        dataKey="inode"
+        data-testId="container-list-table">
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html:55]: Why did you remove the `dot-listing-data-table`? That component already have the `dot-action-header`

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html (hunk 1 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html:55]: The flow we had before involved two different data sources, the table itself and the store, for this reason I use PrimeNG's native table, that way I can keep the data and its manipulation in the store

Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.html:55]: The `dot-listing-data-table` should be a `presentational component` to display the data we passed to it. However, that can be discussed in another `issue`.

**Review by jdotcms — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]
